CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+edition)](https://pubs.opengroup.org/onlinepubs/9699919799/) or later.
+
+## For ddclient Project Maintainers
+
+### Merging Pull Requests
+
+To facilitate reviews and code archaeology, `master` should have a
+semi-linear commit history like this:
+
+```
+*   f4e6e90 sandro.jaeckel@gmail.com 2020-05-31 07:29:51 +0200 (master)
+|\          Merge pull request #142 from rhansen/config-line-format
+| * 30180ed rhansen@rhansen.org 2020-05-30 13:09:38 -0400
+|/          Expand comment documenting config line format
+*   01a746c rhansen@rhansen.org 2020-05-30 23:47:54 -0400
+|\          Merge pull request #138 from rhansen/dyndns-za-net
+| * 08c2b6c rhansen@rhansen.org 2020-05-29 14:44:57 -0400
+|/          Replace dydns.za.net with dyndns.za.net
+*   d65805b rhansen@rhansen.org 2020-05-30 22:30:04 -0400
+|\          Merge pull request #140 from ddclient/fix-interpolation
+| * babbef1 sandro.jaeckel@gmail.com 2020-05-30 04:03:44 +0200
+|/          Fix here doc interpolation
+*   6ae69a1 rhansen@rhansen.org 2020-05-30 22:23:57 -0400
+|\          Merge pull request #141 from ddclient/show-debug-ssl
+| * 096288e sandro.jaeckel@gmail.com 2020-05-30 04:42:27 +0200
+| |         Expand tabs to spaces in vim
+| * 0206262 sandro.jaeckel@gmail.com 2020-05-30 04:40:58 +0200
+|/          Show debug connection settings after evaluating use-ssl
+...
+```
+
+See https://stackoverflow.com/a/15721436 for an explanation of the
+benefits.
+
+To achieve a history like the above, the pull request must be rebased
+onto `master` before merging. Unfortunately, GitHub does not have a
+one-click way to do this (the "Rebase and merge" option does a
+fast-forward merge, which is not what we want). See
+[isaacs/github#1143](https://github.com/isaacs/github/issues/1143) and
+[isaacs/github#1017](https://github.com/isaacs/github/issues/1017). Until
+GitHub adds that feature, it has to be done manually:
+
+```shell
+# Set this to the name of the GitHub user or project that owns the
+# fork used for the pull request:
+PR_USER=
+
+# Set this to the name of the branch in the fork used for the pull
+# request:
+PR_BRANCH=
+
+# The commands below assume that `origin` refers to the
+# ddclient/ddclient repository
+git remote set-url origin git@github.com:ddclient/ddclient.git
+
+# Add a remote for the fork used in the PR
+git remote add "${PR_USER:?}" git@github.com:"${PR_USER:?}"/ddclient
+
+# Fetch the latest commits for the PR and ddclient master
+git remote update -p
+
+# Switch to the pull request branch
+git checkout -b "${PR_USER:?}-${PR_BRANCH:?}" "${PR_USER:?}/${PR_BRANCH:?}"
+
+# Rebase the commits (optionally using -i to clean up history) onto
+# the current ddclient master branch
+git rebase origin/master
+
+# Force update the contributor's fork. This will only work if the
+# contributor has checked the "Allow edits by maintainers" box in the
+# PR. If not, you will have to manually merge the rebased commits.
+git push -f
```

Comment:
Why do you force a push?

Reply:
Rebasing causes divergence, so Git will reject the push unless forced.